pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -163,88 +173,65 @@ func (r *DualReadWriter) CreateResource(ctx context.Context, path string, ref st
 	}
 
 	parsed, err := r.parser.Parse(ctx, info)
-	if err != nil {
-		return nil, fmt.Errorf("parse file: %w", err)
-	}
-
-	data, err = parsed.ToSaveBytes()
 	if err != nil {
 		return nil, err
 	}
 
-	if err := r.repo.Create(ctx, path, ref, data, message); err != nil {
-		return nil, fmt.Errorf("create resource in repository: %w", err)
-	}
-
-	// Directly update the grafana database
-	// Behaves the same running sync after writing
-	// FIXME: to make sure if behaves in the same way as in sync, we should
-	// we should refactor the code to use the same function.
-	if ref == "" {
-		if _, err := r.folders.EnsureFolderPathExist(ctx, path); err != nil {
-			return nil, fmt.Errorf("ensure folder path exists: %w", err)
-		}
-
-		if err := parsed.Run(ctx); err != nil {
-			return nil, fmt.Errorf("run resource: %w", err)
-		}
-	} else {
-		if err := parsed.DryRun(ctx); err != nil {
-			logger := logging.FromContext(ctx).With("path", path, "name", parsed.Obj.GetName(), "ref", ref)
-			logger.Warn("failed to dry run resource on create", "error", err)
-			// Do not fail here as it's purely informational
-			parsed.Errors = append(parsed.Errors, err.Error())
-		}
-	}
-
-	return parsed, nil
-}
-
-// UpdateResource updates a resource in the repository
-func (r *DualReadWriter) UpdateResource(ctx context.Context, path string, ref string, message string, data []byte) (*ParsedResource, error) {
-	if err := repository.IsWriteAllowed(r.repo.Config(), ref); err != nil {
-		return nil, err
+	// Always use the provisioning identity when writing
+	ctx, _, err = identity.WithProvisioningIdentity(ctx, parsed.Obj.GetNamespace())
+	if err != nil {
+		return nil, fmt.Errorf("unable to use provisioning identity %w", err)
 	}
 
-	info := &repository.FileInfo{
-		Data: data,
-		Path: path,
-		Ref:  ref,
+	// Make sure the value is valid
+	if err := parsed.DryRun(ctx); err != nil {
+		logger := logging.FromContext(ctx).With("path", path, "name", parsed.Obj.GetName(), "ref", ref)
+		logger.Warn("failed to dry run resource on create", "error", err)
+		// Do not fail here as it's purely informational
+		parsed.Errors = append(parsed.Errors, err.Error())
 	}
 
-	// TODO: improve parser to parse out of reader
-	parsed, err := r.parser.Parse(ctx, info)
-	if err != nil {
-		return nil, fmt.Errorf("parse file: %w", err)
+	if len(parsed.Errors) > 0 {
```

Comment:
I still find this errors more annoying than useful.

Reply:
these are exactly the errors we want to show in the PR feedback!   We still need to return all the other info like where you are trying to write things etc

Reply:
OK for POST+PUT, i switched it to return 500 (should be 400)

lets see how that feels and can revisit